site/src/docs/configuration/scripts.md

```diff
@@ -126,6 +154,18 @@ fn my_env_test() {
 }
 ```
 
+### Pre-timeout scripts
+
+A given pre-timeout script _S_ is executed when the current profile has at least one rule where the `platform` predicates match the current execution environment, the script _S_ is listed in `pre-timeout`, and a test matching the `filter` has reached its configured timeout.
+
+Pre-timeout scripts are executed serially, in the order they are defined (_not_ the order they're specified in the rules). If any pre-timeout script exits with a non-zero exit code, an error is logged but the test run continues.
+
+Nextest sets the following environment variables when executing a pre-timeout script:
+
+  * **`NEXTEST_PRE_TIMEOUT_TEST_PID`**: the ID of the process running the test.
+  * **`NEXTEST_PRE_TIMEOUT_TEST_NAME`**: the name of the running test.
+  * **`NEXTEST_PRE_TIMEOUT_TEST_BINARY`**: the name of the binary running the test.
```

Comment:
I'd call the last one `BINARY_ID`. Maybe also pass in the components of the binary ID as separate variables for convenience.